src/hooks/useWorldBySlug.ts

```diff
@@ -0,0 +1,68 @@
+import { useFirestore, useFirestoreCollectionData } from "reactfire";
+import Bugsnag from "@bugsnag/js";
+
+import { World } from "api/admin";
+
+import { worldConverter } from "utils/converters";
+import { WithId } from "utils/id";
+
+type UseWorldEditResult = {
+  world?: WithId<World>;
+  isLoaded: boolean;
+};
+
+/**
+ * Hook which will return the world when the slug is provided.
+ * The intention is to be used on the client side, when the world slug is provided in the url.
+ * @param worldSlug
+ * @returns
+ */
+export const useWorldBySlug: (worldSlug?: string) => UseWorldEditResult = (
+  worldSlug
+) => {
+  const firestore = useFirestore();
+
+  const worldsRef = firestore
+    .collection("worlds")
+    .where("slug", "==", worldSlug ?? "")
+    .withConverter(worldConverter);
+
+  const { data: worlds, status } = useFirestoreCollectionData<WithId<World>>(
+    worldsRef,
+    {
+      initialData: undefined,
+    }
+  );
+
+  const isWorldLoaded = status !== "loading";
+
+  if (!worlds?.[0]) {
+    Bugsnag.notify(
+      `World with the following slug: ${worldSlug} does not exist.`,
+      (event) => {
+        event.severity = "warning";
+        event.addMetadata("hooks::useWorldBySlug", {
+          worldSlug,
+        });
+      }
+    );
```

Comment:
[NOTE]
Not sure if worthy of Bugsnag like multiple worlds found since it is going to be quite visible to the user the world doesn't exist, but will probably add numerous warning entries.

Reply:
This is only for dev purposes and getting notified on slack that this thing happened. It will be easier to trace it from there.